### ISSUE TITLE
About ManBigDecimalExt#div

### ISSUE THREAD

**Issue by zhoujiaping** [manifold-systems/manifold#446]
hello, there is a suggestion for ManBigDecimalExt#div.
the origin code is 
```
    public static BigDecimal div(@This BigDecimal thiz, BigDecimal operand) {
        return thiz.divide(operand, RoundingMode.HALF_EVEN);
    }
```
when i run the code
    System.out.println(new BigDecimal("10")/new BigDecimal("4"));
i got the answer 2(expect 2.5).
and i wonder how to set the scale and roudingmode for operate overriding.
below is my solution(contains 2 file)

the 1st:
```
@Extension
public class BigDecimalExt {
    //div2 rename to div
    public static BigDecimal div2(@This BigDecimal x,BigDecimal y){
        BigDecimalScale cfg = BigDecimalScale.get();
        if(cfg == null){
            return x.divide(y);
        }else{
            return x.divide(y,cfg.scale,cfg.mode);
        }
    }

    public static void main(String[] args) {
        System.out.println(BigDecimalScale.withScale(16, RoundingMode.HALF_UP,()->
                new BigDecimal("1").div2(new BigDecimal("3"))
        ));
        //output: 0.3333333333333333
        System.out.println(BigDecimalScale.withScale(16, RoundingMode.HALF_UP,()->
                new BigDecimal("10").div2(new BigDecimal("4"))
        ));
        //output: 2.5000000000000000
        System.out.println(new BigDecimal("10").div2(new BigDecimal("4")));
        //output: 2.5
        System.out.println(new BigDecimal("1").div2(new BigDecimal("3")));
        //output:  Non-terminating decimal expansion; no exact representable decimal result.
    }
}
```
the 2nd:
```
public class BigDecimalScale {
    private static ThreadLocal<BigDecimalScale> tl = new ThreadLocal<>();
    public int scale;
    public RoundingMode mode;
    public BigDecimalScale(int scala, RoundingMode mode){
        this.scale = scala;
        this.mode = mode;
    }
    public static BigDecimal withScale(int scale, RoundingMode mode, Supplier<BigDecimal> fn){
        BigDecimalScale oldCfg = tl.get();
        try{
            tl.set(new BigDecimalScale(scale,mode));
            return fn.get();
        }finally {
            if(oldCfg == null){
                tl.remove();
            }else{
                tl.set(oldCfg);
            }
        }
    }
    public static BigDecimalScale get(){
        return tl.get();
    }
}
```

**Comment by rsmckinney** [comment]
Yikes!  That's a typo in the BigDecimal divide implementation. is should be `divide(operand, MathContext.DECIMAL128)`. I will change that later today (and add better tests!). Thanks for reporting this.

**Comment by rsmckinney** [comment]
Fix available with release 2023.1.6.
